keymanapp/keyman
# change(mac): remove 'Always show OSK' option

Remove the option in the Options pane of the Configuration titled 'Always show on-screen keyboard'.

Instead, automatically track the OSK state (shown or hidden) and apply it whenever Keyman is activated. Also, save the size and location of the OSK window.

Fixes #1523
Fixes #5856 
Fixes #8965 
Fixes #12341 
Fixes #12342 

# User Testing

* GROUP_CURRENT_OS: Use macOS Sonoma
* GROUP_OLD_OS: Use macOS High Sierra (10.13), Mojave (10.14) or Catalina (10.15)

## TEST_ALWAYS_SHOW_OPTION_IS_GONE

1. Open the Configuration window and go to the Options pane
1. Verify that the checkbox labeled 'Always show on-screen keyboard' is gone

## TEST_OSK_REMEMBERS_SHOW_HIDE_STATE_DURING_SESSION

1. Open the OSK while using Keyman
1. Change from Keyman to another input method such as the U.S. Keyboard provided by Apple
1. Confirm that the OSK automatically hides
1. Switch back to the Keyman input method
1. Confirm that the OSK reappears

## TEST_OSK_STABILITY

1. Open the OSK while using Keyman
1. Change between text input applications with OSK open
1. Click on menus in the various applications
1. Confirm that the OSK does not flicker or disappear while switching apps or using menus

## TEST_OSK_REMAINS_HIDDEN_WITH_KEYMAN_INACTIVE

1. Open the OSK while using Keyman
1. Change from Keyman to another input method such as the U.S. Keyboard provided by Apple
1. Switch applications and click on various menus and windows
1. Confirm that the Keyman OSK never flashes on screen while Keyman is inactive

## TEST_OSK_REMEMBERS_SHOW_HIDE_STATE_AFTER_RESTART

1. Open the OSK while using Keyman
1. Restart the machine
1. Switch to the Keyman input method
1. Confirm that the OSK appears

## TEST_OSK_REMEMBERS_SIZE_AND_LOCATION_AFTER_RESTART

1. Open the OSK while using Keyman
1. Change the size and location of the OSK to something
1. Restart the machine
1. Switch to the Keyman input method
1. Confirm that the OSK appears at the same size and location as when last used
